src/components/Mapper/Mapper.jsx

```diff
@@ -0,0 +1,755 @@
+      clearRect(CLEAR_MODE.HIGHLIGHT);
+    }
+    if (previousRectangle.enc !== null) {
+      const selectedEnc = previousRectangle.enc.find(
+        (encLocation) => {
+          return (isLocationExactlyEqual(location, encLocation));
+        }
+      )
+      if (selectedEnc) {
+        clearRect(CLEAR_MODE.ENCOUNTER, selectedEnc);
+      }
+    }
+    if(previousRectangle.select !== null) {
+      clearRect(CLEAR_MODE.SELECT);
+    }
+    drawRect(location, CLEAR_MODE.SELECT); // Change the fill color
+    previousRectangle.select = { x: location.x, y: location.y, width: location.width, height: location.height };
+
+    locationId.current = location.zoneId;
+    setSelectedZone(location.name);
+    setEncounterList(setAllEncounters(location.zoneId));
+    setTrainerList(getTrainersFromZoneName(location.name));
+
+    setFieldItems(getFieldItemsFromZoneID(location.zoneId));
+    setHiddenItems(getHiddenItemsFromZoneID(location.zoneId));
+    setShopItems(getRegularShopItems(location.zoneId));
+    setScriptItems(getScriptItems(location.zoneId));
+    setFixedShops(getFixedShops(location.zoneId));
+    setHeartScaleShop(getHeartScaleShopItems(location.zoneId));
+  };
+
+  function handleMouseMove(event) {
+    if(rect === null) {//Shouldn't happen but let's make sure
+      return setRect(canvasRef.current.getBoundingClientRect());
+    }
+
+    // This system allows mobile users to make use of the mapper too
+    let scrollLeft = window.scrollX || document.documentElement.scrollLeft;
+    let scrollTop = window.scrollY || document.documentElement.scrollTop;
+
+    const mouseX = event.clientX - rect.left + scrollLeft;
+    const mouseY = event.clientY - rect.top + scrollTop;
+
+    const location = getSelectedLocation(mouseX, mouseY);
+    if (location && location.zoneId !== locationId.current) {
+      setHoveredZone(location.name);
+      drawRect(location); // Change the fill color
+    } else if (location && location.zoneId === locationId.current) {
+      // This prevents the selected location from being highlighted by the hover color
+      drawRect(location, CLEAR_MODE.SELECT);
+    } else {
+      if (location && previousRectangle.enc !== null) {
+        const hoveredEnc = previousRectangle.enc.find(
+          (encLocation) => {
+            return (isLocationExactlyEqual(location, encLocation));
+          }
+        )
+        if (hoveredEnc) {
+          clearRect(CLEAR_MODE.ENCOUNTER, hoveredEnc);
+        }
+      }
+      setHoveredZone(null);
+      if (previousRectangle.highlight !== null) {
+        clearRect(CLEAR_MODE.HIGHLIGHT);
+        previousRectangle.highlight = null;
+      }
+    }
+  }
+
+  const handleMouseLeave = () => {
+    // Clear the hovered zone when mouse leaves
+    setHoveredZone(null);
+    if (previousRectangle.highlight !== null) {
+      clearRect(CLEAR_MODE.HIGHLIGHT);
+    }
+    if (previousRectangle.enc !== null) {
+      clearRect(CLEAR_MODE.ENCOUNTER);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
Half of me is wondering if we put canvas into its own react component with these functions, but ultimately it works wither way.  Eventually if we go the ADA route we'll probably have to add focus and blur event listeners as well.  I really like this code :)

Reply:
I appreciate the callout for ADA. I wasn't thinking about it while developing but we should definitely implement that in a future update.

Reply:
I also think that we should move the canvas into its own component. It's not necessary for beta, but its a good idea that we should implement.